docs/docs/contributing/developing-bloqade.md

```diff
@@ -0,0 +1,172 @@
+
+    EmulatorPy["Emulator IR Python"]
+    EmulatorJL["Emulator IR Julia"]
+
+    QuEra["QuEra IR"]
+    Braket["Braket IR"]
+    JuliaEmulator["Bloqade.jl"]
+    PythonEmulator["Python Emulator"]
+
+    Aquila["Aquila"]
+
+    Builder -->|parse| PythonAST
+    PythonAST -->|lower| EmulatorPy
+    PythonAST -->|lower| QuEra
+    PythonAST -->|lower| Braket
+    PythonAST -->|transpile| JuliaAST
+
+    QuEra -->|execute| Aquila
+    Braket -->|execute| Aquila
+
+    JuliaAST -->|lower| EmulatorJL
+    EmulatorPy -->|execute| PythonEmulator
+    EmulatorJL -->|execute| JuliaEmulator
+
+```
+
+## High-Level Builder Representation
+
+When programming Bloqade using the Python API, the user constructs a representation of an analog quantum
+circuit. This representation is a flattened version of
+the actual analog circuit. Flattened means that the user
+input is a linear sequence of operations where the context of neighboring nodes in the sequence
+of instructions can determine the tree. The Bloqade AST describes the actual analog circuit.
+
+## Bloqade AST
+
+The Bloqade AST/ is a representation of a quantum analog circuit for
+Nuetral atom computing. It is a directed acyclic graph (DAG) with nodes
+for different hierarchical levels of the circuit. The base node is the
+`AnalogCircuit` which contains the geometry of the atoms stored as a
+`AtomArragment` or `ParallelRegister` objects. The Other part of the
+circuit is the `Sequence`, which contains the waveforms that describe
+the drives for the Ryberg/Hyperfine transitions of
+each Rydberg atom. Each transition is represented by a `Pulse` including
+a `Field` for the drive's detuning, rabi amplitude, and rabi phase
+. A Field relates the spatial and temporal dependence
+of a drive. The spatial modulates the temporal dependence of the
+waveform. A DAG also describes the `Waveform` object. Finally, we have basic `Scalar` Expressions as well for describing
+The syntax of Real valued continuous numbers.
+
+## Bloqade Compilers
+
+Given a user program expressed as the Bloqade AST, we can target various
+backends by transforming from the Bloqade AST to some other kind of IR.
+For example, when submitting a task to QuEra's hardware, we lower the
+Bloqade AST to the IR that describes a valid program for the hardware.
+
+We also support compiling the Bloqade AST to a more convenient IR for our
+Python emulator backend. This IR is called the Emulator IR. The Emulator IR
+represents the Bloqade AST that is more convenient for lowering
+directly to a state-vector simulation.
+
+We call the "lowering" procedure from the Bloqade AST to the various other
+IR's code generation. The pattern we use for code generation is called
+the "visitor pattern." The visitor pattern traverses a tree
+structure and performs some action on each node while storing an internal
+state. In Bloqade, the transpilling/lowering steps are typically called
+"code generation" and will live in modules named `codegen`.
```

Comment:
```suggestion
We call the "lowering" procedure from the Bloqade AST to the various other
IR's code generation. The pattern we use for code generation is called
the "visitor pattern." The visitor pattern traverses the tree
structure and performs some action on each node while storing an internal
state. In Bloqade, the transpilling/lowering steps are typically called
"code generation" and will live in modules named `codegen`.
```

This paragraph isn't clear to me if I were an outside reader. Does the "lowering" procedure call the other IR's code gens? Is the lowering procedure a direct python method? This paragraph might even be better suited in the beginning to potentially make explicit what we mean by "lowering".

Reply:
Beginning of the Bloqade "Compilers" section that is.

Reply:
Oh yeah this is just poorly worded I will just rewrite it to make it more clear